moveit_ros/planning/planning_scene_monitor/include/moveit/planning_scene_monitor/planning_scene_monitor.h

```diff
@@ -143,6 +144,20 @@ class PlanningSceneMonitor : private boost::noncopyable
                        const std::shared_ptr<tf2_ros::Buffer>& tf_buffer = std::shared_ptr<tf2_ros::Buffer>(),
                        const std::string& name = "");
 
+  /** @brief Constructor
+   *  @param scene The scene instance to maintain up to date with monitored information
+   *  @param rml A pointer to a kinematic model loader
+   *  @param nh external parent NodeHandle
+   *         Ensure that this NodeHandle's CallbackQueue is different from the global queue and that
+   *         it is served by an external spinner!
```

Comment:
This mechanism can also be used to explicitly run in the global queue, if the user wants it that way (we did it for a long time and it's not necessarily a problem).
Can you change this to just state that the Monitors will be updated with the queue of this node handle?